cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/media/class-filter.php

```diff
@@ -522,17 +522,147 @@ public function filter_video_embeds( $html, $id, $attachment ) {
+			$video = '<!-- wp:video {"id":' . $attrs[2][1] . '} --><figure class="wp-block-video"><video autoplay src="' . $attrs[2][0] . '"></video></figure><!-- /wp:video -->';
+	
+			$content = str_replace( array( '<p>' . $shortcodes[0][$i] . '</p>', $shortcodes[0][$i] ), $video, $content );
+		}
+
+		return $content;
+	}
+
+	/**
+	 * Converts classic image to a Gutenberg block.
+	 *
+	 * @param string $content
+	 * 
+	 * @return string
+	 */
+	protected function classic_image_to_block( $content ) {
+		if ( strpos( $content, '<img class=' ) === false ) {
+			return $content;
+		}
+
+		$dom = new \DOMDocument();       
+		
+		// Suppress errors here as we are not supplying a full-fledged
+		// HTML document, which will cause loadHTML to throw errors.
+		@$dom->loadHTML( $content, LIBXML_HTML_NODEFDTD );
+		$imgs = $dom->getElementsByTagName( 'img' );
+
+		foreach ($imgs as $img) {
+			$class_list = $img->getAttribute( 'class' );
+			
+			// Ensure we're dealing with a proper <img> tag that came from Classic Editor
+			if ( false === strpos( $class_list, 'wp-image-' ) || '' === $img->getAttribute( 'width' ) ) {
+				continue;
+			}
+
+			list( $img_id, $size, $align, $json ) = $this->prepare_image_block_classes( $class_list );
+
+			// Prep the <img> node
+			$img_clone = $img->cloneNode( true );
+			$img_clone->removeAttribute( 'width' );
+			$img_clone->removeAttribute( 'height' );
+			$img_clone->setAttribute( 'class', 'wp-image-' . $img_id );
+
+			// Prep the <figure> node
+			$img_fig = $dom->createElement( 'figure' );
+			$img_fig->appendChild( $img_clone );
+			$img_fig->setAttribute( 'class', 'wp-block-image size-' . $size . ( $align ? ' align' . $align : '' ) );
+			
+			// Surround <figure> with block comments
+			$fragment = $dom->createDocumentFragment();
+			$fragment->appendChild( $dom->createComment( ' wp:image ' . $json . ' ' ) );
+			$fragment->appendChild( $img_fig );
+			$fragment->appendChild( $dom->createComment( ' /wp:image ' ) );
+
+			$img->parentNode->replaceChild( $fragment, $img );
+		}
+
+		$front = strpos( $dom->saveHTML(),'<body>' ) + 6;
+		$end   = ( strrpos ($dom->saveHTML(), '</body>' ) ) - strlen( $dom->saveHTML() );
+
+		return substr( $dom->saveHTML(), $front, $end );
+	}
+
+	/**
+	 * Prepare image components which are important for the conversion.
+	 * Fetch: image id, size and alignment
+	 * 
+	 * @param string $class_list The string which represents all classes on this image.
+	 * 
+	 * @return array
+	 */
+	protected function prepare_image_block_classes( $class_list ) {
+		$results = array();
+
+		// Fetch image id
+		preg_match( '/wp-image-([0-9]+)/', $class_list, $matches );
+		$results[] = $id = isset( $matches[1] ) ? (int) $matches[1] : null;
+
+		// Fetch image size
+		$results[] = $size = 'large';
+		if ( false !== strpos( $class_list, 'size-' ) ) {
+			preg_match( '/size-([A-Za-z0-9-_]+)/', $class_list, $matches );
+
+			$all_sizes = get_intermediate_image_sizes();
```

Comment:
@dugajean 

What you're trying to do here is great and it makes sense. However, this function on VIP will return you an [empty array](https://github.com/WordPress/WordPress-Coding-Standards/commit/e7707b2338e7823a2d14cfbdca8bdc88488d6081#diff-1034de57626216004cbb6c6e1707f313R244-R250).

I'd say to be ok to implement the same logic of the [function](https://developer.wordpress.org/reference/functions/get_intermediate_image_sizes/), as `wp_get_additional_image_sizes()` seems to be a safe method.